subprojects/core/src/testFixtures/groovy/org/gradle/util/TestUtil.groovy

```diff
@@ -300,6 +300,17 @@ class TestUtil {
 
         return cause
     }
+
+    static <T extends Throwable> boolean hasCause(Throwable t, Class<T> expectedCauseType) {
```

Comment:
💭 This returns `true` if `t` itself is of type `expectedCauseType`.  I wouldn't expect that from the name - in this case `t` might not or might not itself be *caused by* that type.

Reply:
True, but I intended it to behave like that and I can't find a better name. Do you have any suggestions?

Reply:
I think `isOrIsCausedBy` might be slightly clearer.